platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -257,51 +367,89 @@ public ITable load() {
-                                Cell cell = (Cell) cd;
-                                AbstractType<?> cellType = cell.column().cellValueType();
-                                value = cellType.getSerializer().deserialize(cell.value());
-                            } else {
-                                // Hillview won't process the complex data
-                                throw new RuntimeException("Hillview can't convert complex data");
+    private static void loadColumns(UnfilteredRowIterator partition,
+        List<IAppendableColumn> columns, boolean[] columnToLoad, List<CassandraType> columTypes){
+        Unfiltered unfiltered = partition.next();
+        if (unfiltered instanceof Row) {
+            Row row = (Row) unfiltered;
+            int i = 0;
+            int currentColumn = 0;
+            Object value;
+            IAppendableColumn col;
+            // Extracting the value for each column from a single row
+            for (ColumnData cd : row) {
+                // Filter the column to only load the one marked true by columnToLoad[]
+                if (columnToLoad[i]) {
+                    if (cd.column().isSimple()) {
+                        col = columns.get(currentColumn);
+                        AbstractType<?> cellType = cd.column().cellValueType();
+                        value = cellType.getSerializer().deserialize(((Cell) cd).value());
+                        if(value == null) {
+                            col.appendMissing();
+                        } else {
+                            switch (columTypes.get(i)) {
+                                case STRING:
+                                    col.append(value.toString());
+                                    break;
+                                case CQLSTRING:
+                                    col.append(cellType.getSerializer()
+                                            .toCQLLiteral(((Cell) cd).value()).toString());
+                                    break;
+                                case INT:
+                                    col.append((Integer) value);
+                                    break;
+                                case SMALLINT:
+                                    col.append(((Short) value).intValue());
+                                    break;
+                                case TINYINT:
+                                    col.append(((Byte) value).intValue());
+                                    break;
+                                case VARINT:
+                                    col.append(((BigInteger) value).doubleValue());
+                                    break;
+                                case BIGINT:
+                                    col.append(((Long) value).doubleValue());
+                                    break;
+                                case DECIMAL:
+                                    col.append(((BigDecimal) value).doubleValue());
```

Comment:
add a helper function which does bounds checking and throws if the value is too large